services/navet-ms/lambdas/pollUser.js

```diff
@@ -1,78 +1,106 @@
 /* eslint-disable no-console */
 import to from 'await-to-js';
-import { throwError } from '@helsingborg-stad/npm-api-error-handling';
+import { InternalServerError, UnauthorizedError } from '@helsingborg-stad/npm-api-error-handling';
 
 import config from '../../../config';
 import params from '../../../libs/params';
 import { parseXml, parseErrorMessageFromXML, parseJSON } from '../helpers/parser';
-import { client } from '../helpers/client';
+import getNavetRequestClient from '../helpers/client';
 import { putEvent } from '../../../libs/awsEventBridge';
 import * as request from '../../../libs/request';
-import * as response from '../../../libs/response';
 
-const SSMParams = params.read(config.navet.envsKeyName);
+const NAVET_PARAMS = params.read(config.navet.envsKeyName);
 
-export const main = async event => {
+export async function main(event) {
   const { user } = event.detail;
-  const navetSSMparams = await SSMParams;
-  navetSSMparams.personalNumber = user.personalNumber;
-  const xml = parseXml(navetSSMparams);
-
-  const [err, navetResponse] = await to(requestNavetUser(xml, navetSSMparams));
-  if (err) return response.failure(err);
-  await putEvent(createNavetPollEventDetail(navetResponse), 'NavetPoll', 'navet.poll');
-  return;
-};
-
-function createNavetPollEventDetail(navetData) {
-  const userObj = {
-    personalNumber: navetData.PersonId.PersonNr,
-    firstName: navetData.Namn.Fornamn,
-    lastName: navetData.Namn.Efternamn,
+
+  const [requestNavetUserError, navetUser] = await to(requestNavetUser(user.personalNumber));
+  if (requestNavetUserError) {
+    return console.error('(Navet-ms)', requestNavetUserError);
+  }
+
+  const eventDetail = getNavetPollEventDetail(navetUser);
+  await putEvent(eventDetail, 'NavetPoll', 'navet.poll');
```

Comment:
Suggestion for improvement, have a putNavetPollEvent function, that has the scope of creating and putting the event tot the event bridge.